src/gdb/GDBDebugSessionBase.ts

```diff
@@ -352,6 +352,15 @@ export abstract class GDBDebugSessionBase extends LoggingDebugSession {
         );
     }
 
+    protected warnAsyncDisabled() {
+        this.sendEvent(
+            new OutputEvent(
+                'The target does not support gdbAsync. Most debugger interactions will only work while the target is stopped.',
+                'important'
```

Comment:
```suggestion
                'The target does not support gdbAsync. Most debugger interactions will only work while the target is stopped.\n'
```

How does this render for you in the Debug Console?
For me it's
<img width="874" height="103" alt="image" src="https://github.com/user-attachments/assets/415ccf43-1f06-4945-9133-0178ead66dcd" />

If I change to the above, it gets (note the linebreak, without it the empty line before the gdb disclaimer misses)
<img width="847" height="98" alt="image" src="https://github.com/user-attachments/assets/ebed0a46-c996-4202-a419-879e5e019182" />

Also, it seems I use a slightly different gdb flavour on my Windows machine. That supports async and I had to provoke the output by changing the returned feature array in a Typescript debug session of the adapter.

Reply:
I admit I had only tested this with verbose mode on, in which case the line breaks in the debug console are a bit of a mess anyway. Thanks for checking this.

I can confirm your observation, but I am not sure your suggestion is an improvement. I deliberately send the message in the `'important'` category, instead of the default `'console'` one as your suggestion does, so that there is a GUI notification, for users who don’t have the Debug Console open. Would you prefer not to have that?

It does seem a bit inconsistent that VS Code renders `important` in the same color as `stdout` (meant for output from the debuggee) and not in the same color as `console` (meant for output from the debugger). But I would rather not work around that if it incurs other costs.

I am a bit puzzled by the `Breakpoint 1 at 0x…` line in your screenshots – the check for async support is supposed to come before setting breakpoints (the `initialized` event in between should enforce this order). If this is reproducible for you, could I have a full log with `verbose` enabled in the launch configuration?

> Also, it seems I use a slightly different gdb flavour on my Windows machine. That supports async

Is it version 13 or newer? That would explain that. It’s only the older versions that don’t support it, and only when launching/attaching directly – connecting via `gdbserver` supported it before.

Reply:
One correction:
I must have had a different GDB in the path when I wasn't able to provoke the problem initially. Retrying with 11.2 and an extension build from a branch including your changes shows it without meddling with variables in the TS debugger.

I still see the breakpoint hit before the message. Note that I set a breakpoint in `initCommands`. Here my launch config
```
        {
            "type": "gdb",
            "request": "launch",
            "name": "gdb launch",
            "cwd": "${workspaceFolder}",
            "gdb": "gdb",
            "gdbAsync": true,
            "verbose": true,
            "program": "xxxxxxxx\\cdt-gdb-adapter\\src\\integration-tests\\test-programs\\loopforever.exe",
            "initCommands": [
                "break main"
            ]
        }
```

And here a zip with logs with and without the change: [logs.zip](https://github.com/user-attachments/files/22425837/logs.zip)

Reply:
> Note that I set a breakpoint in `initCommands`.

Ah, that’s fine then. Nothing unexpected in that case. I was just worried about the adapter setting breakpoints and making decisions about pausing before it had confirmed async support.